sites/funmooc/src/frontend/js/utils/api/web-analytics/index.ts

```diff
@@ -0,0 +1,31 @@
+import type { Maybe } from 'richie-education/js/types/utils';
+import { WebAnalyticsAPI, WebAnalyticsAPIBackend } from 'richie-education/js/types/WebAnalytics';
+import context from 'richie-education/js/utils/context';
+import { handle } from 'richie-education/js/utils/errors/handle';
+import GoogleAnalyticsApi from 'richie-education/js/utils/api/web-analytics/google_analytics';
+import GoogleTagManagerApi from 'richie-education/js/utils/api/web-analytics/google_tag_manager';
+import FunXitiApi from '../../../../../../js/utils/api/web-analytics/xiti';
+
+const WEB_ANALYTICS_PROVIDER = context?.web_analytics_provider;
+
+enum FunMoocWebAnalyticsAPIBackend {
+  XITI = 'xiti',
+}
+
+const WebAnalyticsAPIHandler = (): Maybe<WebAnalyticsAPI> => {
+  try {
+    switch (WEB_ANALYTICS_PROVIDER) {
+      case WebAnalyticsAPIBackend.GOOGLE_ANALYTICS:
+        return new GoogleAnalyticsApi();
+      case WebAnalyticsAPIBackend.GOOGLE_TAG_MANAGER:
+        return new GoogleTagManagerApi();
+      case FunMoocWebAnalyticsAPIBackend.XITI:
+        return new FunXitiApi();
+    }
+  } catch (error) {
+    handle(error);
+  }
+  return undefined;
+};
```

Comment:
I find this weird. Since you are setting `FunMoocWebAnalyticsAPIBackend` to XITI, why not directly return FunXitiApi()?

Reply:
You are right. I'll change it.